packages/alfa-rules/src/sia-r87/rule.ts

```diff
@@ -61,6 +64,11 @@ export default Rule.Atomic.of<Page, Document, Question>({
             )
           );
 
+        // there can be more than one element with a role of main, going to any of these is OK.
+        const mains = document
+          .inclusiveDescendants({ flattened: true, nested: true })
```

Comment:
```suggestion
          .inclusiveDescendants({ flattened: true })
```

References cannot cross document boundaries.